Neos.ContentRepository.StructureAdjustment/src/StructureAdjustmentService.php

```diff
@@ -97,7 +98,14 @@ public function fixError(StructureAdjustment $adjustment): void
             $remediation = $adjustment->remediation;
             $eventsToPublish = $remediation();
             assert($eventsToPublish instanceof EventsToPublish);
-            $this->eventPersister->publishEvents($eventsToPublish)->block();
+
+            $enrichedEventsToPublish = $eventsToPublish->withCausationOfFirstEventAndAdditionalMetaData(
+                EventMetadata::fromArray([
+                    'structureAdjustment' => mb_strimwidth($adjustment->render() , 0, 250, '…')
+                ])
+            );
```

Comment:
IMO this should look something like:
```suggestion
            $enrichedEventsToPublish = $eventsToPublish->withCorrelationId(CorrelationId::fromString(UuidFactory::create());
```